### PR TITLE
README improvements

### LINKED ISSUE #7: Improve documentation - building
How to build?

### PR DESCRIPTION
**Pull request by bennyz**
Fixes #7

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,31 @@
...
+test:
+```
+go test ./...
+```
+
 *Work in progress*
 ## Usage:
 ```
-  ./example-finder search text [-l|--lang LANG] [-t|--token TOKEN]
+./example-finder search TEXT [-l|--lang LANG] [-t|--token TOKEN] [-m|--mode MODE] [-r|--results num]
```
Comment (matanshukry) [inline README.md]: add an example on how to use it

File: README.md [hunk 2]
```diff
@@ -1,8 +1,31 @@
 # example-finder
 
+## Build:
+Dependencies:
+```
+gcc (required for sqlite3 support)
```
Comment (matanshukry) [inline README.md]: give a link on where to download it

Comment (matanshukry) [inline README.md:6]: give a link on where to download it

File: README.md (hunk 1 above)
Comment (matanshukry) [inline README.md:28]: add an example on how to use it

**Review by matanshukry — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (bennyz) [reply README.md:28]: Done

File: README.md (hunk 2 above)
Comment (bennyz) [reply README.md:6]: Done

File: README.md [hunk 3]
```diff
@@ -1,8 +1,40 @@
...
 ```
-  ./example-finder search text [-l|--lang LANG] [-t|--token TOKEN]
+./example-finder search TEXT [-l|--lang LANG] [-t|--token TOKEN] [-m|--mode MODE] [-r|--results num]
+  lang    - language to return results in (not implemented yet)
+  token   - Github API token, see https://github.com/settings/tokens - should be inside .token file
+  mode    - currently only REST mode is supported, a future plan is to support GraphQL as well
+  results - number of results per page to return (default 30)
+
+Example usage:
+./example-finder search go-pg -r 100 -t ABC
```
Comment (matanshukry) [inline README.md:35]: explain -t/token with example. you're saying it's a token, yet talking about file in description. is ABC the entire token? is it the name of the file? If there is no way to pass a file, specify it's taken from .token file if not given (assuming it is correct behavior)

**Review by matanshukry — CHANGES_REQUESTED** [review]

**Pull request merged**